lib/between_meals/knife.rb

```diff
@@ -80,6 +93,20 @@ def cookbook_upload(cookbooks)
       end
     end
 
+    def berks_cookbook_upload(cookbooks)
+      # cookbooks: array
+      # cookbook_paths: array
+      if cookbooks.any?
+        @cookbook_dirs.each do |path|
+          cookbooks.each do |cb|
+            @logger.warn("Running berkshelf on cookbook: #{cb}")
```

Comment:
you have:
chef/cookbooks/special/
chef/cookbooks/other/
chef/cookbooks/anotherkind/

each has a subset of your cookbooks. Right now, you're walking deach directory and then trying to upload a changed cookbook, so you'll try to upload special/mycookbook other/mycookbook and anotherkind/mycookbook, only one of which will exist.